src/pages/LogIn/components/LoginForm.tsx

```diff
@@ -12,106 +13,106 @@ interface IPropsFromDispatch {
   setAuthToken: typeof setAuthToken;
 }
 
-interface IState {
-  usernameValue: string;
-  passwordValue: string;
-  isRememberMeChecked: boolean;
-  showHelperText: boolean;
-  isValidUsername: boolean;
-  isValidPassword: boolean;
-}
-
 type AllProps = IPropsFromDispatch & RouteComponentProps;
-
-class LoginFormComponent extends React.Component<AllProps, IState> {
-  constructor(props: AllProps) {
-    super(props);
-    this.state = {
-      usernameValue: "",
-      passwordValue: "",
-      isRememberMeChecked: true,
-      showHelperText: false,
-      isValidUsername: true,
-      isValidPassword: true,
-    };
-    this.handleSubmit = this.handleSubmit.bind(this);
-  }
-
-  // Description: Create a fake user to work with API, redux store
-  async handleSubmit(event: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
+const LoginFormComponent : React.FC<AllProps>=({
+  setAuthToken,
+}: AllProps)=>{
+  const [usernameValue,setUsernameValue]=React.useState<string>("");
+  const [passwordValue,setPasswordValue]=React.useState<string>("");
+  const [isRememberMeChecked,setIsRememberMeChecked]=React.useState<boolean>(true);
+  const [showHelperText,setShowHelperText]=React.useState<boolean>(false);
+  const [isValidUsername,setIsValidUsername]=React.useState<boolean>(true);
+  const [isValidPassword,setIsValidPassword]=React.useState<boolean>(true);
+  const [errorMessage,setErrorMessage]=React.useState<string>("");
+  const history = useHistory();
+  async function handleSubmit(event: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
     event.preventDefault();
-    const { setAuthToken } = this.props;
     const authURL = `${process.env.REACT_APP_CHRIS_UI_AUTH_URL}`;
     let token;
-    try {
-      token = await ChrisApiClient.getAuthToken(
+
+    if(!usernameValue){
+      setIsValidUsername(false);
+    }
+    if(!passwordValue){
+      setIsValidPassword(false);
+    }
+    else{
+      setIsValidUsername(true);
+      setIsValidPassword(true);
+
+      try {
+        token = await ChrisApiClient.getAuthToken(
         authURL,
-        this.state.usernameValue,
-        this.state.passwordValue
+        usernameValue,
+        passwordValue
       );
     } catch (error) {
-      this.setState({
-        showHelperText: true,
-      });
+      console.log({error});
```

Comment:
Seeing https://github.com/FNNDSC/ChRIS_ui/pull/247/commits/961ab97d533959d4cfb0a5e68286229cbdfedc0d, is your choice to not use `console.log`? If so, then delete this one too.